gems/sorbet-runtime/lib/types/_types.rb

```diff
@@ -148,8 +148,11 @@ def self.cast(value, type, checked: true)
   # doesn't match the type.
   def self.let(value, type, checked: true)
     return value unless checked
-
-    Private::Casts.cast(value, type, "T.let")
+    if Configuration.recursively_check_in_tests?
+      Private::Casts.cast_recursive(value, type, "T.let")
+    else
+      Private::Casts.cast(value, type, "T.let")
+    end
```

Comment:
This degrades performance of `let` for something that we only ever intend to exist for one-off debugging.